libcudacxx/include/cuda/std/detail/libcxx/include/__utility/pair.h

```diff
@@ -69,707 +70,704 @@
-                    is_nothrow_constructible<second_type, _U2&&>::value))
-        : first(_CUDA_VSTD::forward<_U1>(__p.first)), second(_CUDA_VSTD::forward<_U2>(__p.second)) {}
-#endif // defined(__cuda_std__) && !defined(__CUDACC_RTC__)
-
-#if _CCCL_STD_VER > 2020
-    template<class _U1, class _U2, __enable_if_t<
-            _CheckArgs::template __is_pair_constructible<const _U1&&, const _U2&&>::value
-    >* = nullptr>
-    _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_INLINE_VISIBILITY constexpr
-    explicit(!_CheckArgs::template __is_implicit<const _U1&&, const _U2&&>::value)
-    pair(const pair<_U1, _U2>&& __p)
-        noexcept(is_nothrow_constructible<first_type, const _U1&&>::value &&
-                 is_nothrow_constructible<second_type, const _U2&&>::value)
-        : first(_CUDA_VSTD::move(__p.first)), second(_CUDA_VSTD::move(__p.second)) {}
-
-#if defined(__cuda_std__) && !defined(__CUDACC_RTC__)
-    template<class _U1, class _U2, __enable_if_t<
-            _CheckArgs::template __is_pair_constructible<const _U1&&, const _U2&&>::value
-    >* = nullptr>
-    _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_HOST constexpr
-    explicit(!_CheckArgs::template __is_implicit<const _U1&&, const _U2&&>::value)
-    pair(const ::std::pair<_U1, _U2>&& __p)
-        noexcept(is_nothrow_constructible<first_type, const _U1&&>::value &&
-                 is_nothrow_constructible<second_type, const _U2&&>::value)
-        : first(_CUDA_VSTD::move(__p.first)), second(_CUDA_VSTD::move(__p.second)) {}
-#endif // defined(__cuda_std__) && !defined(__CUDACC_RTC__)
-#endif // _CCCL_STD_VER > 2020
+struct __pair_constraints
+{
+  static constexpr bool __implicit_default_constructible =
+    __is_implicitly_default_constructible<_T1>::value && __is_implicitly_default_constructible<_T2>::value;
+
+  static constexpr bool __explicit_default_constructible =
+    !__implicit_default_constructible && _LIBCUDACXX_TRAIT(is_default_constructible, _T1)
+    && _LIBCUDACXX_TRAIT(is_default_constructible, _T2);
+
+  static constexpr bool __implicit_copy_constructible_from_T =
+    _LIBCUDACXX_TRAIT(is_copy_constructible, _T1) && _LIBCUDACXX_TRAIT(is_copy_constructible, _T2)
+    && _LIBCUDACXX_TRAIT(is_convertible, __make_const_lvalue_ref<_T1>, _T1)
+    && _LIBCUDACXX_TRAIT(is_convertible, __make_const_lvalue_ref<_T2>, _T2);
+
+  static constexpr bool __explicit_copy_constructible_from_T =
+    _LIBCUDACXX_TRAIT(is_copy_constructible, _T1) && _LIBCUDACXX_TRAIT(is_copy_constructible, _T2)
+    && (!_LIBCUDACXX_TRAIT(is_convertible, __make_const_lvalue_ref<_T1>, _T1)
+        || !_LIBCUDACXX_TRAIT(is_convertible, __make_const_lvalue_ref<_T2>, _T2));
+
+  static constexpr bool __implicit_move_constructible_from_T =
+    _LIBCUDACXX_TRAIT(is_move_constructible, _T1) && _LIBCUDACXX_TRAIT(is_move_constructible, _T2)
+    && _LIBCUDACXX_TRAIT(is_convertible, __make_const_lvalue_ref<_T1>, _T1)
+    && _LIBCUDACXX_TRAIT(is_convertible, __make_const_lvalue_ref<_T2>, _T2);
```

Comment:
`__make_const_lvalue_ref` looks very wrong here, since we are checking moves. This should probably just use `_T1, _T1` like in the explicit version of this below, right?